### PR TITLE
Fix calling Add._from_args and Mul._from_args

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136143

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -386,6 +386,38 @@ def canonicalize_bool_expr(expr: SympyBoolean) -> SympyBoolean:
...
+        if cls is sympy.Add:
+            sort_fn = sympy.core.add._addsort
+        elif cls is sympy.Mul:
+            sort_fn = sympy.core.mul._mulsort
+        else:
+            raise ValueError(f"Unknown cls: {cls}")
+
+        # we don't support non commutative with sort
+        assert is_commutative is True
+        args = list(args)
```
Comment (Skylion007) [inline torch/fx/experimental/symbolic_shapes.py:408]: nit: args should already be a list or there is a type error lol

**Review by ezyang — APPROVED** [review]
How are you finding these? Do we need more tests for internal sympy invariants?

**Comment by isuruf** [comment]
> How are you finding these? Do we need more tests for internal sympy invariants?

After https://github.com/pytorch/pytorch/pull/136079, I realized that `_from_args` does less things than I thought it did. Just read all the PRs where I added _from_args, to make sure I didn't have any assumptions.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Pull request closed without merging**